### PR TITLE
Add <pokemonicon> and <itemicon>

### PR DESCRIPTION
**Pull request by bumbadadabum**
I talked most of this through with Zarel, just leaving this here first as a sanity check. Usage of this feature will be added once this is merged.

### CODE REVIEW

File: js/battledata.js [hunk 1]
```diff
@@ -786,7 +786,18 @@ var Tools = {
 			return {attribs: attribs};
 		};
 		return function (input) {
-			return html.sanitizeWithPolicy(getString(input), tagPolicy);
+			var str = getString(input);
```
Comment (Zarel) [inline js/battledata.js]: I'd just reuse `input` instead of creating a new variable for it...

**Comment by bumbadadabum** [comment]
Updated.

**Comment by Slayer95** [comment]
This is a job that can be perfectly done with the Caja API :|

**Comment by Slayer95** [comment]
Just whitelist these tag names and return ``{tag, attribs}`` with the rewritten tag from the ``tagPolicy``.

**Comment by Slayer95** [comment]
Actually, that's not the case if we read the inner HTML of the ``<pokemonicon>`` tag rather than its attributes. Is there a reason for doing it that way rather than with attributes?

**Comment by Zarel** [comment]
I told him it was probably possible with the Caja API! He decided to do it this way so he didn't need to learn the Caja API.

**Comment by Slayer95** [comment]
Heh. I can refactor this now. Would you want any attribute name in particular?
I'd suggest ``<psicon pokemon="heracross"></psicon>`` and ``<psicon item="leppaberry"></psicon>``

**Comment by Zarel** [comment]
There's no other reason not to do it with attributes, but it _is_ shorter and more readable currently.

**Comment by Slayer95** [comment]
My first suggestion is longer just for 1 character for Pokémon and shorter by 4 for items.
If you care enough for length so to skip the attribute quotes, then it's shorter for both cases.

Using attributes also improves extensibility as I already said, prevents doing multiple passes over the HTML code, and of course the correcteness sense of just using the HTML sanitizer module for the HTML jobs.

**Comment by Zarel** [comment]
I think you miscounted, it's longer by 1 for Pokémon and longer by 8 for items. But more importantly I think it's more readable/intuitive this way.

And technically, transforming custom tags _is_ a different job from sanitizing. Putting them in different parts of the codebase may be good for readability.

These are all really minor reasons, though. I don't have a strong opinion either way. @bumbadadabum?

**Comment by Slayer95** [comment]
Playing with HTML sanitizing is never intuitive enough.

Before
```
> Tools.sanitizeHTML('<div id="<pokemonicon>heracross</pokemonicon>"</div>')
<"<div id="&lt;pokemonicon&gt;heracross&lt;/pokemonicon&gt;"></div>"
```

After

```
> Tools.sanitizeHTML('<div id="<pokemonicon>heracross</pokemonicon>"</div>')
< "<div id="&lt;span class=" style="display: inline-block ; background: transparent url(&#34;//play.pokemonshowdown.com/sprites/smicons-sheet.png?a1&#34;) no-repeat scroll -400px -510px">"</div>"
```

This commit definitely makes it harder to reason about how we handle HTML.

**Comment by Slayer95** [comment]
cc @kotarou3 @xfix

**Comment by Zarel** [comment]
That's a fair criticism.

**Comment by bumbadadabum** [comment]
@Slayer95 I'm fine with a refactor obviously. I did mention to @Zarel I did it in the easiest, "dirtiest" way I could think of. Your PR does seem more "proper" so yeah that's great.

**Pull request merged**